apps/client/src/hooks/usePermissionedGroups.ts

```diff
@@ -35,12 +32,10 @@ export default function usePermissionedGroups(): ReturnParameters {
     )
 
     const generateIdentityCommitment = useCallback(
-        async (signer: Signer, groupName: string): Promise<string | null> => {
-            setLoading(true)
-            const identity = await createIdentity(
-                (message) => signer.signMessage(message),
-                groupName
-            )
+        async (groupName: string): Promise<string | null> => {
+            const nonce = 0
+            const message = `Sign this message to generate your ${groupName} Semaphore identity with key nonce: ${nonce}.`
+            const identity = new Identity(message)
```

Comment:
Imo, This generates deterministic identity for public messages.

Reply:
Yep, that massage should be signed by users with their private keys and Metamask